### ISSUE TITLE
Link to PDF not showing in performance record

### ISSUE THREAD

**Issue by kelea99** [pulibrary/dpul#809]
Question: Is this a bug or is this something that will need its own ticket for a feature/enhancement to the application?

Example item with link to PDF that was added in DPUL and is not working in DPUL: https://dpul.princeton.edu/performances/catalog/a538af11-8d21-4be7-aaa0-40c1e9514d89

Use Case: Dan from Mendel would like to include links to PDFs of performance programs in each item for Performance Ensembles Collection (these links, when available, are also in the online catalog record for the audio performance). 

I see two possibilities here: 
1. Dan may add these manually in DPUL via the DPUL-created medatada field.
2.  We map the MARC field containing the PDF link (in the 856...which poses its own obvious problems, I would think) to figgy, and then to DPUL.

Thoughts?

**Comment by tpendragon** [comment]
We already pull the 856s and put them in the metadata in Figgy. There must be some bug with the metadata sync for PUO. It used to work.

**Comment by kelea99** [comment]
hmm. weird.ok. changing to bug.

**Comment by tpendragon** [comment]
I think this is just a bug in DPUL. I see the value in Figgy just fine.

**Comment by hackmastera** [comment]
I don't see the link in figgy @tpendragon. Here is the figgy record: https://figgy.princeton.edu/catalog/a538af11-8d21-4be7-aaa0-40c1e9514d89.

The click-throughs to the catalog are working but the link presented in the catalog gives a 404, which seems like another issue. Do we know where this program is?

**Comment by tpendragon** [comment]
@hackmastera I should have been clearer. The link is in the JSON-LD which gets synced to DPUL. See "electronic_locations": https://figgy.princeton.edu/catalog/a538af11-8d21-4be7-aaa0-40c1e9514d89.jsonld

**Comment by hackmastera** [comment]
@tpendragon thanks. So it's still giving a 404; have these been moved but not updated in voyager? @kelea99?

**Comment by tpendragon** [comment]
https://dpul.princeton.edu/performances/catalog/9407e8a3-b844-4d32-8da5-1d94c94c37b8 is another example.

**Comment by hackartisan** [comment]
The feature was added in https://github.com/pulibrary/dpul/pull/738

**Comment by hackartisan** [comment]
The problem is that the field key from figgy is "Electronic locations" but the custom field key in the exhibit was changed to "Program PDF" so it doesn't find the field in custom fields so it doesn't index it.

**Comment by hackartisan** [comment]
I'm not sure there's a straightforward code fix here. For other fields where curators didn't like the label that came from figgy it looks like we've edited them globally.

@kelea99 is it acceptable to keep the field called "Electronic locations"? Or, is there another field name that would be acceptable (generally for all 856es in all collations) that we could put in on the backend? The link text will still reference the fact that it's a program.

![Screen Shot 2022-07-27 at 3 03 37 PM](https://user-images.githubusercontent.com/845363/181351915-42eebc91-ec1b-4bea-8f2e-4dc00427cbd9.png)

**Comment by kelea99** [comment]
Suggestion: "Available Online" is what the online catalog uses for this category. That would cover anything in that field.

**Comment by hackartisan** [comment]
If stakeholders are agreeable, deploying https://github.com/pulibrary/dpul/pull/1365 and reindexing the collection will apply that change.

**Comment by hackartisan** [comment]
A note about what I discovered in the code.

metadata values are transformed into a label => value hash and the original jsonld key is discarded as part of preparing the data for indexing.

https://github.com/pulibrary/dpul/blob/8bcdcca21321e3a982e6b3a32b4450fe83aa21ad/app/values/manifest_metadata.rb#L39

The lookup on custom fields is later, and uses the custom field label as opposed to the slug:

https://github.com/pulibrary/dpul/blob/8bcdcca21321e3a982e6b3a32b4450fe83aa21ad/app/services/iiif_manifest.rb#L114

There may be background as to why it uses label. I think there was a problem with duplicate custom fields at one point.

I tried to use the custom field slug by passing it back with the metadata hash embedded in the value, but that messed up the code path that doesn't go through jsonld.

It may help to fully separate the jsonld case from the manifest case by using different objects.
